pkg/build/api/v1/types.go

```diff
@@ -542,17 +611,18 @@ type BuildConfig struct {
 
 // BuildConfigSpec describes when and how builds are created
 type BuildConfigSpec struct {
-	// triggers determine how new Builds can be launched from a BuildConfig. If no triggers
-	// are defined, a new build can only occur as a result of an explicit client build creation.
-	Triggers []BuildTriggerPolicy `json:"triggers"`
-
 	// RunPolicy describes how the new build created from this build
 	// configuration will be scheduled for execution.
 	// This is optional, if not specified we default to "Serial".
 	RunPolicy BuildRunPolicy `json:"runPolicy,omitempty"`
 
-	// BuildSpec is the desired build specification
-	BuildSpec `json:",inline"`
+	//Triggers determine how new Builds can be launched from a BuildConfig. If
+	//no triggers are defined, a new build can only occur as a result of an
+	//explicit client build creation.
+	Triggers []BuildTriggerPolicy `json:"triggers"`
+
+	// CommonSpec is the desired build specification
+	CommonSpec `json:",inline"`
```

Comment:
This should be at the top, just like in the internal.  Don't reorder fields unless they have to.